src/etc/gdb_providers.py

```diff
@@ -85,6 +85,31 @@ def to_string(self):
     def display_hint():
         return "string"
 
+class StdSliceProvider:
+    def __init__(self, valobj):
+        self.valobj = valobj
+        self.length = int(valobj["length"])
+        self.data_ptr = valobj["data_ptr"]
+
+    def to_string(self):
+        return "{}(size={})".format(self.valobj.type, self.length)
+
+    def children(self):
+        for index in xrange(self.length):
+            element_ptr = self.data_ptr + index
+            try:
+                # rust-lang/rust#64343: passing deref expr to `str` allows
+                # catching exception on garbage pointer
+                str(element_ptr.dereference())
+                yield "[{}]".format(index), element_ptr.dereference()
+            except RuntimeError:
+                yield str(index), "inaccessible"
+
+                break
```

Comment:
Am I correct in assuming that this `break` will be hit after yielding the first `"inaccessible"` -- and thus is an alternative approach to the `saw_inaccessible` check done in the `StdVecProvider`? If so, I would prefer if this was implemented the same way in both cases. Otherwise the inconsistency is confusing.